src/core/coap/coap_secure.cpp

```diff
@@ -70,12 +75,13 @@ Error CoapSecure::Start(uint16_t aPort)
 
 Error CoapSecure::Start(MeshCoP::SecureTransport::TransportCallback aCallback, void *aContext)
 {
-    Error error = kErrorNone;
+    Error error = kErrorAlready;
 
     mConnectedCallback.Clear();
-
-    SuccessOrExit(error = mDtls.Open(&CoapSecure::HandleDtlsReceive, &CoapSecure::HandleDtlsConnected, this));
-    SuccessOrExit(error = mDtls.Bind(aCallback, aContext));
+    SuccessOrExit(mDtls.SetMaxConnectionAttempts(0, nullptr, nullptr));
+    SuccessOrExit(mDtls.Open(HandleDtlsReceive, HandleDtlsConnected, this));
+    SuccessOrExit(mDtls.Bind(aCallback, aContext));
```

Comment:
In the other `Start()` method above, we store the returned error, but here we don't. What's the difference?

Reply:
Just to make the returned `error` match the documentation. We want `kErrorNone` or `kErrorAlready`. From `SetMaxConnectionAttempt()` we may return `kErrorInvalidState`.

I see your point, they both behave correctly but look different. Let me change them both to be consistent.
Will use the model of setting `error = kErrorNone` at the end (this sounds better as we won't assume/rely on error types that may be returned from `mDtls` calls).

Reply:
Pushed the update.
I actually added a new private `Open()` method which is then used from different `Start()` methods and does the common part of preparing and openning the `mDtls` socket.

Reply:
Nice! I was thinking the same thing too.